### PR TITLE
Deprecate file and blob

### PR DESCRIPTION
**Pull request by BelSasha**

### CODE REVIEW

**Comment by BelSasha** [comment]
* **#1213** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1213?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BelSasha and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/provenance.py [hunk 1]
```diff
@@ -10,15 +10,14 @@
...
-from runhouse.resources.blobs import file
 
 # Need to alias so it doesn't conflict with the folder property
 from runhouse.resources.folders import Folder, folder as folder_factory
 from runhouse.resources.hardware import _current_cluster, _get_cluster_from, Cluster
 from runhouse.resources.resource import Resource
 from runhouse.rns.top_level_rns_fns import resolve_rns_path
 from runhouse.rns.utils.api import log_timestamp, resolve_absolute_path
-from runhouse.utils import StreamTee
+from runhouse.utils import StreamTee, write_data_to_a_file
```
Comment (jlewitt1) [inline runhouse/resources/provenance.py:20]: actually i think if we're killing provenance soon we don't need to bother with this refactor - if `write_data_to_a_file` isn't used anywhere else we can prob remove also

File: runhouse/resources/secrets/provider_secrets/azure_secret.py [hunk 2]
```diff
@@ -47,30 +45,19 @@ def _write_to_file(
...
-                    data = ss.read()
-                path.write(data, serialize=False, mode="w")
-            else:
-                full_path = os.path.expanduser(path)
-                Path(full_path).parent.mkdir(parents=True, exist_ok=True)
-                with open(full_path, "w") as f:
-                    parser.write(f)
-                new_secret._add_to_rh_config(path)
+            full_path = os.path.expanduser(path)
+            Path(full_path).parent.mkdir(parents=True, exist_ok=True)
```
Comment (jlewitt1) [inline runhouse/resources/secrets/provider_secrets/azure_secret.py:49]: maybe we put these lines in a helper if we're using in a few places?

File: runhouse/resources/provenance.py (hunk 1 above)
Comment (jlewitt1) [reply runhouse/resources/provenance.py:20]: might be easiest to add this PR to the stack for removing `provenance` instead of against main

**Review by carolineechen — APPROVED** [review]

**Comment by rohinb2** [comment]
### Merge activity

* **Sep 3, 2:01 PM EDT**: @rohinb2 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/1213).

**Pull request merged**